Upgrade to Spring Boot 3.4.7

fhanik opened the pull request.

fhanik committed:
Upgrade to Spring Boot 3.4.6

fhanik committed:
Test changes for Spring Boot Upgrade

fhanik committed:
More behavior fixes

fhanik committed:
Add back missing dependency

strehle commented on server/src/main/java/org/cloudfoundry/identity/uaa/util/UaaHttpRequestUtils.java:
```diff
@@ -201,15 +202,28 @@ public UaaHttpRequestRetryHandler(int executionCount) {
+
+        @Override
+        public boolean retryRequest(HttpResponse response, int execCount, HttpContext context) {
+            if (executionCount > this.executionCount) {
```
```suggestion
            if (executionCount > this.executionCount) {
```
```suggestion
            if (execCount > this.executionCount) {
```

strehle commented on dependencies.gradle:
```diff
@@ -120,9 +121,9 @@ libraries.springWeb = "org.springframework:spring-web:${versions.springFramework
 libraries.statsdClient = "com.timgroup:java-statsd-client:3.1.0"
 libraries.thymeleafDialect = "nz.net.ultraq.thymeleaf:thymeleaf-layout-dialect:3.4.0"
-libraries.thymeleafExtrasSpringSecurity5 = "org.thymeleaf.extras:thymeleaf-extras-springsecurity5:3.1.3.RELEASE"
+libraries.thymeleafExtrasSpringSecurity5 = "org.thymeleaf.extras:thymeleaf-extras-springsecurity6:3.1.3.RELEASE"
```
thymeleafExtrasSpringSecurity5 maybe remove the version or increase to 6 ?

strehle commented on server/build.gradle:
```diff
@@ -109,6 +110,10 @@ dependencies {
     testImplementation(libraries.awaitility)
 
     implementation(libraries.commonsIo)
+    implementation "commons-codec:commons-codec:1.17.2"
```
can we move this back to dependency.gradle

duanemay committed:
Clean up dependencies

Remove more of the explicit versions, float with spring boms
spring-security to 6.4.7
jackson to 2.19.1
greenmail to 2.1.3

duanemay committed:
Fix for ErrorRoutingIT

fhanik committed:
Test pollution after Spring upgrade?

strehle committed:
Refactor AutoLoginIT

* create a new user for the simpleAutologinFlow
* ensure with new user to run and see the approval page

strehle committed:
Fix CSRF check in the same session

if /oauth/authorize is requested but internal forward to /oauth/confirm_access is done, then no further check is needed

duanemay committed:
Refactor AutologinIT better message on missing cookies

Instead of  expected: 2 but was: 1

duanemay committed:
Update Docs request parameter references

duanemay committed:
Upgrade Tomcat Cargo container from 9.x to 10.x

fhanik commented:
@strehle @duanemay Feel free to merge when you are ready!

fhanik committed:
Sometimes the rest template does not have the right credentials. Bypass it.

strehle committed:
update boot

strehle committed:
Merge remote-tracking branch 'origin/develop' into pr/2025-spring-boot-upgrade

strehle committed:
Merge remote-tracking branch 'origin/develop' into pr/2025-spring-boot-upgrade

# Conflicts:
#	server/src/main/java/org/cloudfoundry/identity/uaa/security/web/SecurityFilterChainPostProcessor.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/ErrorRoutingIT.java

strehle committed:
test build error

strehle committed:
fix recursive call

strehle commented:
I have here a cleanup PR, https://github.com/fhanik/uaa/pull/2 

Please check.

The JARs in WEB-INF/lib show http5 and http4 , need to check if this is really needed, but then I would merge this PR

strehle committed:
Cleanup before Spring Boot 3 upgrade

duanemay committed:
Clean up library references

- Remove unreferenced deps
- Remove duplicate braveInstrumentation
- Rename jakartaValidationApi from javaxValidationApi
- Uses libraries.lombok instead of full dep ref

duanemay reviewed (approved):
👍
